images/github-runners-manager/manager.py

```diff
@@ -0,0 +1,107 @@
+                        runner_name = f"ggml-runner-{workflow.id}-{job.id}-{workflow_run.event}-{int(time.time())}"
+
+                        print(f"TRIGGERING {runner_name} for workflow_name={workflow.name}")
+                        runner_dir = f'/runners/{runner_name}'
+                        os.makedirs(runner_dir)
+                        os.chmod(runner_dir, 0o755)
+                        os.chown(runner_dir, 1000, 1000)
+
+                        # Get a JIT runner config
+                        jitrequest = {
+                            'name': runner_name,
+                            'runner_group_id': 1,  # FIXME what to put here
+                            'labels': ["self-hosted", "X64", "linux", *args.runner_label],
+                            'work_folder': "/tmp/github-runner/_work"
+                        }
+                        response = requests.post(
+                            f" https://api.github.com/repos/{args.repo}/actions/runners/generate-jitconfig",
+                            headers={
+                                'Authorization': f'Bearer {args.token}',
+                                'X-GitHub-Api-Version': "2022-11-28"
+                            },
+                            json=jitrequest)
+                        if response.status_code != 201:
+                            print(f"invalid JIT response code: {response.status_code}\n    {response.text}")
+                            continue
+                        jitconfig = response.json()
+
+                        # start the worker in its container and wait for finish
+                        print(
+                            f"Running job runner id={jitconfig['runner']['id']} os={jitconfig['runner']['os']} labels={[value['name'] for value in jitconfig['runner']['labels']]}")
+                        try:
+                            client.containers.run("ggml-github-runner", jitconfig['encoded_jit_config'],
+                                                  entrypoint="/entrypoint.sh",
+                                                  name=runner_name,
+                                                  runtime="nvidia",
+                                                  device_requests=[
+                                                      DeviceRequest(device_ids=["all"],
+                                                                    capabilities=[['gpu']])],
+                                                  user='1000:1000',
+                                                  auto_remove=True,
+                                                  tmpfs={
+                                                      f'/mnt/runners/{runner_name}/tmp': 'size=8G,uid=1000'
+                                                  },
+                                                  volumes={
+                                                      f'/mnt/runners/{runner_name}': {'bind': '/tmp/github-runner',
```

Comment:
Yeah we only need to mount bind if github runner requires to do so. If not, we should prevent bind mount because created files will not be cleaned automatically.

Reply:
For better security, we can also add `security_opt=["no-new-privileges:true"]` to prevent it from using other UID/GID than 1000:1000

Reply:
> Yeah we only need to mount bind if github runner requires to do so. If not, we should prevent bind mount because created files will not be cleaned automatically.

@ggerganov reminds me that ggml-ci avoids downloading models at every run, so we would probably need one host volume finally.

Reply:
If we need to somehow reuse the model, I'd suggest downloading them on host (maybe via `./start-github-runner-manager.sh`), then mount the host directory to container as read-only.

Reply:
> security_opt=["no-new-privileges:true"]

I did not managed to start prometheus with this flag:
- https://docs.kernel.org/userspace-api/no_new_privs.html
- https://github.com/phymbert/llama.cpp/actions/runs/8416990703/job/23044617796

So I disable for now, if you have an idea to fix it, welcomed